fdbserver/storageserver.actor.cpp

```diff
@@ -1611,6 +1612,67 @@ void merge(Arena& arena,
 	}
 }
 
+ACTOR Future<Optional<Value>> quickGetValue(StorageServer* data, StringRef key, Version version) {
+	if (data->shards[key]->isReadable()) {
+		try {
+			// TODO: Use a lower level API may be better? Or tweak priorities?
+			GetValueRequest req(Span().context, key, version, Optional<TagSet>(), Optional<UID>());
+			data->actors.add(data->readGuard(req, getValueQ));
+			GetValueReply reply = wait(req.reply.getFuture());
+			return reply.value;
+		} catch (Error& e) {
+			std::cout << "quickGetValue fallback because of exception " << e.name() << std::endl;
+		}
+	} else {
+		std::cout << "quickGetValue fallback because not in the server" << std::endl;
+	}
+
+	state Transaction tr(data->cx);
+	tr.setVersion(version);
+	// TODO: is DefaultPromiseEndpoint the best priority for this?
+	tr.info.taskID = TaskPriority::DefaultPromiseEndpoint;
```

Comment:
this means the second read go through the read path front door. 

Can we craft something like how you get the first read?
```
GetValueRequest req(Span().context, key, version, Optional<TagSet>(), Optional<UID>());
 			data->actors.add(data->readGuard(req, getValueQ));
 			GetValueReply reply = wait(req.reply.getFuture());
```

If the second read is also on the same SS, you will save the overhead of getKeyLocation() and the serialization and deserialization on the SS.

Reply:
If it is in the same SS, it should have returned here https://github.com/apple/foundationdb/pull/5609/files#diff-c4dbfae809f757f2061099ee5021ce2cbbf0f4dbc2f0f18480a4856d63d033ceR1623. Note https://github.com/apple/foundationdb/pull/5609/files#diff-c4dbfae809f757f2061099ee5021ce2cbbf0f4dbc2f0f18480a4856d63d033ceR1630-R1637 is a fallback when the data is not in current SS.

Reply:
The first query's code is at `getKeyValuesAndHopQ` which is effectively same as a normal range scan aka `getKeyValuesQ`.

`hop()` issues the second queries from the results returned by the first query.

`quickGetValue` is the second query.